scalafix-core/shared/src/main/scala/scalafix/rule/Rule.scala

```diff
@@ -0,0 +1,191 @@
+package scalafix
+package rule
+
+import scala.meta._
+import scalafix.internal.config.MetaconfigPendingUpstream
+import scalafix.internal.config.ScalafixMetaconfigReaders
+import scalafix.internal.config.ScalafixConfig
+import scalafix.syntax._
+import metaconfig.Conf
+import metaconfig.ConfDecoder
+import metaconfig.Configured
+
+/** A Scalafix Rule is a checker/linter/rewrite/auto-fixer.
+  *
+  * To implement a rewrite/auto-fixer, override the `fix` method. Example:
+  * {{{
+  *   object ReverseNames extends Rule("ReverseNames") {
+  *     override def fix(ctx: RuleCtx) =
+  *       ctx.tree.collect {
+  *         case name @ Name(value) => ctx.replaceTree(name, value.reverse)
+  *       }.asPatch
+  *   }
+  * }}}
+  *
+  * To implement a linter/checker, override the `check` method. Example:
```

Comment:
same as above, I would pick a name and stick to it instead of providing alternatives `linter/checker`

Reply:
Agreed, went with "To report violations of this rule (without automatic fix)"